dsfs/dataset.go

```diff
@@ -186,13 +186,24 @@ func CreateDataset(store cafs.Filestore, ds *dataset.Dataset, df cafs.File, pk c
 		log.Debug(err.Error())
 		return
 	}
-	df, _, err = prepareDataset(store, ds, df, pk)
+
+	if !(dsPrev == nil || dsPrev.IsEmpty()) {
```

Comment:
Apply DeMorgan's Law and make this:

`if dsPrev != nil && !dsPrev.IsEmpty() {`